config/xp_role.go

```diff
@@ -17,181 +23,446 @@ type XpRole struct {
-		err := resp.SetMessage("Le rôle n'a pas été trouvé dans la config.").Send()
+	_, role := cfg.FindXpRoleID(id)
+	if role == nil {
+		err = session.InteractionAPI().Respond(i.Interaction, &interaction.Response{
+			Type: types.InteractionResponseChannelMessageWithSource,
+			Data: &interaction.ResponseData{
+				Flags:   channel.MessageFlagsEphemeral,
+				Content: "Impossible de trouver le rôle. Peut-être a-t-il été supprimé ?",
+			},
+		})
 		if err != nil {
-			s.LogError(err, "role not found")
+			session.LogError(err, "Sending Unable to get role message")
 		}
 		return
 	}
-	r.XP = configModifyMap[getKeyConfigRole(i)]
-	err := gokord.DB.Save(r).Error
+	response := &interaction.Response{
+		Type: types.InteractionResponseModal,
+		Data: &interaction.ResponseData{
+			Title:    "Modification du niveau lié au rôle",
+			CustomID: fmt.Sprintf("xp_role_edit_level_%d", id),
+			Components: []component.Component{
+				&component.Label{
+					Label: "Nouveau niveau",
+					Component: &component.TextInput{
+						Style:       component.TextInputShort,
+						Required:    true,
+						CustomID:    "level",
+						MinLength:   1,
+						MaxLength:   5,
+						Placeholder: "5",
+						Value:       strconv.FormatUint(uint64(exp.Level(role.XP)), 10),
+					},
+				},
+			},
+		},
+	}
+	err = session.InteractionAPI().Respond(i.Interaction, response)
+	if err != nil {
+		session.LogError(err, "Sending Edit level modal")
+	}
+}
+
+func HandleXpRoleEditLevel(
+	session *discordgo.Session,
+	i *discordgo.InteractionCreate,
+	data interaction.ModalSubmitData,
+	parameters []string,
+	resp *cmd.ResponseBuilder,
+) {
+	id, err := getRoleLevelID(parameters)
 	if err != nil {
-		s.LogError(err, "saving config for role %s in %s", roleId, i.GuildID)
+		session.LogError(err, "Reading dynamic CustomID")
 		return
 	}
-	if err = resp.SetMessage("Rôle modifié.").Send(); err != nil {
-		s.LogError(err, "sending role saved")
+
+	fmt.Printf("Alors?... %#v", data.Components)
+	levelInput := data.Components[0].(*component.Label).Component.(*component.TextInput)
+	level, err := strconv.Atoi(levelInput.Value)
+	if err != nil || level < 0 {
+		err = resp.IsEphemeral().
+			SetMessage(
+				fmt.Sprintf("Le niveau doit être un nombre entier positif.\n-# Trouvé : %s", levelInput.Value),
+			).
+			Send()
+		if err != nil {
+			session.LogError(err, "Sending bad number warning message")
+		}
+		return
 	}
-}
+	xp := exp.LevelXP(uint(level))
 
-func HandleXpRoleDel(_ *discordgo.Session, _ *discordgo.InteractionCreate, _ interaction.MessageComponentData, resp *cmd.ResponseBuilder) {
-	//err := resp.IsEphemeral().
-	//	SetMessage("Rôle à supprimer").
-	//	SetComponents(component.New().Add(component.NewActionRow().Add(component.NewRoleSelect(XpRoleDelRole)))).
-	//	Send()
-	//if err != nil {
-	//	logger.Alert("config/xp_reduce.go - Sending response to del", err.Error())
-	//}
+	cfg := GetGuildConfig(i.GuildID)
+	for _, xpRole := range cfg.XpRoles {
+		if xpRole.XP == xp {
```

Comment:
Do we really want to restrict this? This does not cause a bug

Reply:
I was lazy and didn't check the code to make sure it can handle this case. I'm also not sure if it is needed to prevent having the same role at different levels, I could remove that as well?